src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -791,20 +792,19 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
-  g_ceph_context->_conf->set_val_or_die("pid_file", "");
+
+  vector<const char*> temp_args(args);
+  md_config_t().parse_argv(temp_args);
 
```

Comment:
@liupan1111 

1. Yes
2. I mean to move env_to_vec() to do_map.

env_to_vec is necessary but only for global context initialization. When parsing specific rbd-nbd command line arguments we should not use arguments provided via environment.

So, in rbd_nbd() you create and fill `args` using only argv_to_vec() and then filter it out with mg_config_t::parse_argv(). In do_map() you create and fill `args` using both arv_to_vec and env_to_vec.